documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -0,0 +1,176 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'prometheus',
+        rules: [
+          {
+            alert: 'PrometheusBadConfig',
+            expr: |||
+              prometheus_config_last_reload_successful{%(prometheusSelector)s} == 0
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'Prometheus failed to reload config, see container logs',
+            },
+          },
+          {
+            alert: 'PrometheusNotificationQueueRunningFull',
+            expr: |||
+              predict_linear(prometheus_notifications_queue_length{%(prometheusSelector)s}[5m], 60 * 30)
```

Comment:
This should be indented by 2

Reply:
The jsonnet language doesn't support that; the first line indentation dictates the rest of the block.  So we can't support your style guide I'm afraid.

Reply:
||| doesn't support it, jsonnet does if you use any of the other forms of string: https://jsonnet.org/ref/spec.html

Reply:
It's not that easy.

The following is legal jsonnet:
```jsonnet
            expr: '
                predict_linear(prometheus_notifications_queue_length{%(prometheusSelector)s}[5m], 60 * 30)
              >
                prometheus_notifications_queue_capacity{%(prometheusSelector)s}
            ' % $._config,
```

However, if you run `jsonnetfmt` on it, it will transform it into something quite unreadable. But even if you do not run `jsonnetfmt` on it, then the same unreadable form will still end up in the YAML file:
```yaml
    "expr": "\n                predict_linear(prometheus_notifications_queue_length{job=\"prometheus\"}[5m], 60 * 30)\n              >\n                prometheus_notifications_queue_capacity{job=\"prometheus\"}\n            "
```

What we would need is something like YAML's `|2`. But that's missing from jsonnet.

The only way out I could come up with is the following:
```jsonnet
            expr: |||
              # (indentation hint)
                predict_linear(prometheus_notifications_queue_length{%(prometheusSelector)s}[5m], 60 * 30)
              >
                prometheus_notifications_queue_capacity{%(prometheusSelector)s}
            ||| % $._config,
```

If nobody has a better idea, I'll go with that one.

Reply:
Or how about this one?
```jsonnet
            expr: |||
              (
                predict_linear(prometheus_notifications_queue_length{%(prometheusSelector)s}[5m], 60 * 30)
              >
                prometheus_notifications_queue_capacity{%(prometheusSelector)s}
              )
            ||| % $._config,
```

Reply:
I went for the parentheses method as it appears the least invasive and most readable. Sad that jsonnet's block quoting is not at par with yaml's one here.
I'll resolve this conversation, but feel free to come up with other ideas.